frontend/lib/src/components/widgets/DataFrame/columns/cells/LinkCell.tsx

```diff
@@ -0,0 +1,174 @@
+  readonly displayText?: string | null
+}
+
+export type LinkCell = CustomCell<LinkCellProps>
+
+function onClickSelect(
+  e: Parameters<NonNullable<CustomRenderer<LinkCell>["onSelect"]>>[0]
+): string | null | undefined {
+  const canvas = document.createElement("canvas")
+  const ctx = canvas.getContext("2d", { alpha: false })
+  if (ctx === null) return
+
+  const { posX: hoverX, bounds: rect, cell, theme } = e
+  const font = `${theme.baseFontStyle} ${theme.fontFamily}`
+  ctx.font = font
+
+  const { href, displayText } = cell.data
+
+  const rectHoverX = rect.x + hoverX
+
+  const textWidth = ctx.measureText((displayText || href) ?? "").width
+  const textStart = rect.x + theme.cellHorizontalPadding
+
+  const isHovered =
+    rectHoverX > textStart && rectHoverX < textStart + textWidth
+
+  if (isHovered) {
+    return href
+  }
+
+  return undefined
+}
+
+export const linkCellRenderer: CustomRenderer<LinkCell> = {
+  draw: (args, cell) => {
+    const { ctx, rect, theme, hoverX = -100, highlighted } = args
+    const { href, displayText } = cell.data
+    if (isNullOrUndefined(href)) return
+
+    const displayValue = displayText || href
+
+    const xPad = theme.cellHorizontalPadding
+
+    const drawX = rect.x + xPad
+
+    const rectHoverX = rect.x + hoverX
+
+    const font = `${theme.baseFontStyle} ${theme.fontFamily}`
+    ctx.font = font
+    const middleCenterBias = getMiddleCenterBias(ctx, font)
+    const drawY = rect.y + rect.height / 2 + middleCenterBias
+
+    const isHovered = rectHoverX > rect.x && rectHoverX < rect.x + rect.width
+
+    if (isHovered) {
+      // draw the underline only when the cell is hovered
+      const metrics = measureTextCached(displayValue, ctx, font)
+
+      ctx.moveTo(drawX, Math.floor(drawY + UNDERLINE_OFFSET) + 0.5)
+      ctx.lineTo(
+        drawX + metrics.width,
+        Math.floor(drawY + UNDERLINE_OFFSET) + 0.5
+      )
+
+      ctx.strokeStyle = theme.linkColor
+      ctx.stroke()
+
+      ctx.fillStyle = highlighted
+        ? blend(theme.accentLight, theme.bgCell)
+        : theme.bgCell
+      ctx.fillText(displayValue, drawX - 1, drawY)
+      ctx.fillText(displayValue, drawX + 1, drawY)
+
+      ctx.fillText(displayValue, drawX - 2, drawY)
+      ctx.fillText(displayValue, drawX + 2, drawY)
+    }
+
+    ctx.fillStyle = theme.linkColor
+    ctx.fillText(displayValue, drawX, drawY)
+    ctx.closePath()
+
+    return true
+  },
+  isMatch: (c): c is LinkCell =>
+    (c.data as LinkCellProps).kind === "link-cell",
+  kind: GridCellKind.Custom,
+  measure: (ctx, cell, theme) => {
+    const { href, displayText } = cell.data
+    if (isNullOrUndefined(href)) return 0
+
+    return (
+      ctx.measureText(displayText || href).width +
+      theme.cellHorizontalPadding * 2
+    )
+  },
+  needsHover: true,
+  needsHoverPosition: true,
+  onSelect: e => {
+    const redirectLink = onClickSelect(e)
+
+    if (!isNullOrUndefined(redirectLink)) {
+      window.open(redirectLink, "_blank")
```

Comment:
nit: to have this behave the same way as the link in the overlay editor, it might be good to add `noopener,noreferrer` here as well:

```suggestion
      window.open(redirectLink, "_blank", "noopener,noreferrer")
```